### PR TITLE
Extend network commands with new fields

### PR DESCRIPTION
**Pull request by GokceGK**

### CODE REVIEW

File: internal/cmd/beta/network/create/create.go [hunk 1]
```diff
@@ -113,11 +135,20 @@ func NewCmd(p *print.Printer) *cobra.Command {
 
 func configureFlags(cmd *cobra.Command) {
 	cmd.Flags().StringP(nameFlag, "n", "", "Network name")
-	cmd.Flags().StringSlice(ipv4DnsNameServersFlag, []string{}, "List of DNS name servers for IPv4")
+	cmd.Flags().StringSlice(ipv4DnsNameServersFlag, []string{}, "List of DNS name servers for IPv4. Nameservers cannot be defined for routed networks.")
 	cmd.Flags().Int64(ipv4PrefixLengthFlag, 0, "The prefix length of the IPv4 network")
-	cmd.Flags().StringSlice(ipv6DnsNameServersFlag, []string{}, "List of DNS name servers for IPv6")
+	cmd.Flags().String(ipv4PrefixFlag, "", "The IPv4 prefix of the network (CIDR)")
+	cmd.Flags().String(ipv4GatewayFlag, "", "The IPv4 gateway of a network. If not specified, the first IP of the network will be assigned as the gateway. If 'null' is sent, then the network doesn't have a gateway.")
```
Comment (joaopalet) [inline internal/cmd/beta/network/create/create.go:141]: This should be adjusted here, for no network we have the `--no-ipv4-network` flag

File: internal/cmd/beta/network/create/create.go [hunk 2]
```diff
@@ -113,11 +135,20 @@ func NewCmd(p *print.Printer) *cobra.Command {
...
+	cmd.Flags().StringSlice(ipv6DnsNameServersFlag, []string{}, "List of DNS name servers for IPv6. Nameservers cannot be defined for routed networks.")
 	cmd.Flags().Int64(ipv6PrefixLengthFlag, 0, "The prefix length of the IPv6 network")
+	cmd.Flags().String(ipv6PrefixFlag, "", "The IPv6 prefix of the network (CIDR)")
+	cmd.Flags().String(ipv6GatewayFlag, "", "The IPv6 gateway of a network. If not specified, the first IP of the network will be assigned as the gateway. If 'null' is sent, then the network doesn't have a gateway.")
```
Comment (joaopalet) [inline internal/cmd/beta/network/create/create.go:145]: Same here

File: internal/cmd/beta/network/create/create.go [hunk 3]
```diff
@@ -113,11 +135,20 @@ func NewCmd(p *print.Printer) *cobra.Command {
...
+	cmd.Flags().Bool(routedFlag, false, "If set to true, the network is routed and therefore accessible from other networks")
+	cmd.Flags().Bool(noIpv4Gateway, false, "If set to true, the network doesn't have an IPv4 gateway.")
```
Comment (joaopalet) [inline internal/cmd/beta/network/create/create.go:147]: Soma flag descriptions are ending in `.` and other not. I think usually we end them without punctuation across other commands

File: internal/cmd/beta/network/list/list.go [hunk 4]
```diff
@@ -151,15 +151,20 @@ func outputResult(p *print.Printer, outputFormat string, networks []iaas.Network
...
 			publicIp := ""
 			if network.PublicIp != nil {
 				publicIp = *network.PublicIp
 			}
 
-			table.AddRow(*network.NetworkId, *network.Name, *network.State, publicIp)
+			routed := false
+			if network.Routed != nil {
+				routed = *network.Routed
+			}
```
Comment (joaopalet) [inline internal/cmd/beta/network/list/list.go:165]: :+1:

File: internal/cmd/beta/network/create/create.go (hunk 3 above)
Comment (GokceGK) [reply internal/cmd/beta/network/create/create.go]: Removed the dots from the sentence endings

File: internal/cmd/beta/network/create/create.go (hunk 1 above)
Comment (GokceGK) [reply internal/cmd/beta/network/create/create.go]: edited description

File: internal/cmd/beta/network/create/create.go [hunk 5]
```diff
@@ -50,12 +64,20 @@ func NewCmd(p *print.Printer) *cobra.Command {
...
+				`$ stackit beta network create --name network-1 --no-ipv4-gateway`,
+			),
+			examples.NewExample(
+				`Create an IPv4 network with name "network-1" with DNS name servers, a prefix, a gateway and a prefix length`,
+				`$ stackit beta network create --name network-1  --ipv4-dns-name-servers "1.1.1.1,8.8.8.8,9.9.9.9" --ipv4-prefix-length 25 --ipv4-prefix "10.1.2.0/24" --ipv4-gateway "10.1.2.3"`,
+			),
+			examples.NewExample(
+				`Create an IPv6 network with name "network-1" with DNS name servers, a prefix, a gateway and a prefix length`,
+				`$ stackit beta network create --name network-1  --ipv6-dns-name-servers "2001:4860:4860::8888,2001:4860:4860::8844" --ipv6-prefix-length 56 --ipv6-prefix "2001:4860:4860::8888" --ipv6-gateway "2001:4860:4860::8888"`,
 			),
```
Comment (joaopalet) [inline internal/cmd/beta/network/create/create.go:81]: For some Terraform provider example, the IaaS colleagues said that providing a prefix and a prefix length is redundant, and the prefi length will be ignored if prefix is set. So we should adjust the examples also here

File: internal/cmd/beta/network/list/list.go (hunk 4 above)
Comment (joaopalet) [reply internal/cmd/beta/network/list/list.go:165]: This same handling could also be applied to the output of the `describe`, right?

File: internal/cmd/beta/network/create/create.go (hunk 5 above)
Comment (GokceGK) [reply internal/cmd/beta/network/create/create.go]: updated the examples

File: internal/cmd/beta/network/list/list.go (hunk 4 above)
Comment (GokceGK) [reply internal/cmd/beta/network/list/list.go:165]: I adapted the routed row handling

File: internal/cmd/beta/network/create/create_test.go [hunk 6]
```diff
@@ -155,27 +171,69 @@ func TestParseInput(t *testing.T) {
...
+		},
+		{
+			description: "use routed true",
+			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
+				flagValues[nonRoutedFlag] = "false"
+			}),
+			isValid: true,
+			expectedModel: fixtureInputModel(func(model *inputModel) {
+				model.NonRouted = false
 			}),
```
Comment (joaopalet) [inline internal/cmd/beta/network/create/create_test.go:237]: This is aleady the default, right?

Comment (joaopalet) [reply internal/cmd/beta/network/create/create_test.go:237]:
```suggestion
			description: "non-routed network",
			flagValues: fixtureFlagValues(func(flagValues map[string]string) {
				flagValues[nonRoutedFlag] = "true"
			}),
			isValid: true,
			expectedModel: fixtureInputModel(func(model *inputModel) {
				model.NonRouted = true
			}),
```

File: internal/cmd/beta/network/create/create_test.go
Comment (joaopalet) [inline internal/cmd/beta/network/create/create_test.go:1]: Missing some more unit tests for the `buildRequest` function

File: internal/cmd/beta/network/describe/describe.go [hunk 7]
```diff
@@ -152,6 +152,19 @@ func outputResult(p *print.Printer, outputFormat string, network *iaas.Network)
 			table.AddSeparator()
 		}
 
+		routed := true
+		if network.Routed != nil {
+			routed = *network.Routed
+		}
```
[the remark this answers is not in the document]
Comment (GokceGK) [reply internal/cmd/beta/network/describe/describe.go:158]: normally it should not be possible to have routed field empty. But somehow I managed to create a broken network in my project and in there routed field did not have any value and cli had NPE in list and describe commands. That is why I added this control.

File: internal/cmd/beta/network/create/create_test.go
Comment (GokceGK) [reply internal/cmd/beta/network/create/create_test.go:1]: added some more unit tests

File: internal/cmd/beta/network/describe/describe.go (hunk 7 above)
[the remark this answers is not in the document]
Comment (joaopalet) [reply internal/cmd/beta/network/describe/describe.go:158]: Alright. But then the question is just if for some reason the `routed` field is not present in the response, I think we should either show `false` or not show the value in the output. Showing `true` might be misleading

[the remark this answers is not in the document]
Comment (GokceGK) [reply internal/cmd/beta/network/describe/describe.go]: you are right, I set them back to false. That was also the initial idea of implementation

**Review by vicentepinto98 — APPROVED** [review]

**Pull request merged**